p2p/host/basic/addrs_manager.go

```diff
@@ -114,6 +145,15 @@ func (a *addrsManager) Start() error {
 			return fmt.Errorf("error starting addrs reachability tracker: %s", err)
 		}
 	}
+	if a.observedAddrsManager != nil {
+		a.observedAddrsManager.Start()
```

Comment:
As mentioned in our call, this should take a Notifier interface so the obsaddr manager can register its Notifee. And the obsaddr worker should be move to the obsaddr manager.